commit/solvers.py

```diff
@@ -284,22 +284,23 @@ def fista( y, A, At, tol_fun, tol_x, max_iter, verbose, x0, omega, proximal) :
     """
 
     # Initialization
-    res = -y.copy()
+    
```

Comment:
Are we sure about these modifications?

Reply:
Yes.
The variable `res` was initialized as: `res = -y.copy()`, 
then updated with: `res += A.dot(xhat)`, without using it inbetween. 
I put the operations together and add the weigthed multiplication: `res = confidence_map * (A.dot(xhat) -y.copy()) `